### PR TITLE
Add an option to newContext to disable memoization of global context …

### PR DESCRIPTION
**Pull request by danlapid**
…template.

For Python Isolate Pools we want to instantiate emscripten in a separate context on isolate creation (before we have compatibility flags) and later after receiving the compatibility flags create a new context. 
One blocker to this approach is all the templates that are instantiated in the first context are memoized. 
This PR allows us to create a new context without memoizing the global context templates. 
This feature is off by default and will be enabled in a future PR for python isolates only.

### CODE REVIEW

File: src/workerd/jsg/resource.h
Comment (jasnell) [inline src/workerd/jsg/resource.h:537]: Non-blocking nit... really not a big fan of bool args like this. Much prefer enums for this type of case. Won't block on it tho

File: src/workerd/jsg/resource.h
Comment (jasnell) [inline src/workerd/jsg/resource.h:1557]: Can this be `if constexpr (memoize) {`?

Comment (danlapid) [reply src/workerd/jsg/resource.h:1557]: yep, no difference as compiler will optimize it away but I generally like the if constexpr feature so lets use it 😄

**Comment by danlapid** [comment]
@jasnell is this good to go?

File: src/workerd/jsg/resource.h
Comment (dom96) [inline src/workerd/jsg/resource.h:1470]: Probably a stupid question, but I wonder why we cannot overwrite the memoized context after it has been memoized?

**Review by dom96 — COMMENTED** [review]
Changes LGTM assuming this is the cleanest way to achieve what we need. I don't have a lot of context on this code so will leave to someone else to approve.

File: src/workerd/jsg/resource.h
Comment (danlapid) [reply src/workerd/jsg/resource.h:1470]: That would require the inverse of the same logic changing the code from that existed before this change rather than adding a different code flow for our exceptional case.

**Review by jasnell — APPROVED** [review]
LGTM but there's a merge conflict and let's be sure to run an internal CI job before merging to make sure there are no issues on the internal repo

**Pull request merged**
